### ISSUE TITLE
Wartezeiten sowie länge der Grün und Rotphasen einstellen können

### ISSUE THREAD

**Issue by Damian-Rutkowski** [Andreas-Kreuz/ak-lua-bibliothek-fuer-eep#31]
Ich zähle die Fahrzeuge über Kontaktpunkte, mir fiel nachts bei wenig Verkehr auf das die Kreuzungen keine Abweichungen haben, sie werden alle nach gleicher Wartezeit geschaltet. Irgendwo ist da schon sowas wie eine "Grünewelle" möglich, aber sollte das nicht absichtlich sein dann fehlt mir irgendwie die Abwechslung. Sicherlich würde das schon reichen wenn man einstellen könnte wie lange so eine Grünphase / Rotphase dauert bei Phase x.

Ist sowas in Zukunft vorgesehen?

**Comment by Andreas-Kreuz** [comment]
Meine Idee an der Stelle wäre, für jede der Schaltungen eine "Grünzeit" konfigurierbar zu machen. Ist das die gewünschte Funktion?

**Comment by Damian-Rutkowski** [comment]
Ja, das halte ich sogar für besser, als an den Ampeln das zu vergeben, weniger Fehlverhalten und die Zeile verlängert sich nicht noch extra.

Ich halte jedoch weiter darin fest auch eine „Rotzeit“ einzuführen. Besonders wenn Kreuzungen Groß sind oder bei engen Kurven brauchen die Fahrzeuge länger, oder wenn Linksabbieger die Kreuzung räumen müssen da diese warten mussten bis der Gegenverkehr rot bekommen hat.

**Comment by Andreas-Kreuz** [comment]
Klingt nach einer "Räume-die-Kreuzung-Zeit", "Wartezeit-vor-Grün" bzw. "Wartezeit-zwischen-den-Schaltungen".
Um es nicht zu kompliziert werden zu lassen würde ich folgende Dinge umsetzen:
* pro Schaltung: Länge der Grünphase pro Schaltung
* pro Kreuzung: Länge der Wartezeit nach einer Schaltung

**Comment by Damian-Rutkowski** [comment]
Ich find das gar nicht kompliziert, ich hoffe wir reden vom selben.
Es wäre nicht schlecht an der Funktion „kx:fuegeSchaltungHinzu(kx_schaltungx, rotphase=10, gruenphase=7)“ einstellen zu können.
